sphinx/texinputs/footnotehyper-sphinx.sty

```diff
@@ -153,6 +169,21 @@
                             \protect\footnotemark[#1]\fi}%
 \AtBeginDocument % let hyperref less complain
    {\pdfstringdefDisableCommands{\def\sphinxfootnotemark [#1]{}}}%
+% to obtain hyperlinked footnotes in longtable environment we must replace
+% hyperref's patch of longtable's patch of \@footnotetext by our own
+\def\sphinxlongtablepatch {% only for longtable wrapped in "savenotes"
+    \def\LT@p@ftntext
+    {\ifx\FNH@endfntext@next\FNH@endfntext@nolink
+     \expandafter\FNH@fn@fntext\else\expandafter\FNH@hyper@fntext\fi }%
```

Comment:
hmm.. the "False" branch is always taken due to misformed test. But this is correct for Sphinx anyhow. The test (from a dev version of upstream footnotehyper) was to avoid a `\@footnotetext` trying to do an hyperlink when none exists do to `\footnotemark[N]/\footnotetext[N]` case. But Sphinx uses environment form for `footnotetext` and never encounters `\@footnotetext` then. Hence the misformed test is not a bug for Sphinx. Nevertheless it is a bug which I will fix after getting feedback from upstream footnotehyper.